shared/middleware/assets_api.rb

```diff
@@ -59,16 +31,10 @@ def self.s3
   #
   get %r{/v3/assets/([^/]+)/([^/]+)$} do |encrypted_channel_id, filename|
     dont_cache
-    not_found unless type = filename.split('.').last
+    not_found unless (type = filename.split('.').last)
```

Comment:
Please consider splitting this across two lines to avoid the side effect in the conditional; e.g.
type = filename.split(.').last
not_found if type.nil?

(Side effects in conditionals are a frequent source of error or misunderstanding; e.g. confusing = and ==, and it's more important to optimize for readability than to save a line.)